internal/cmd/root.go

```diff
@@ -38,28 +46,50 @@ var rootCmd = &cobra.Command{
 It provides an interactive chat interface with AI capabilities, code analysis, and LSP integration
 to assist developers in writing, debugging, and understanding code directly from the terminal.`,
 	Example: `
-# Run in interactive mode
-crush
+	# Run in interactive mode
+	crush
+
+	# Run with debug logging
+	crush -d
+
+	# Run with debug logging in a specific directory
+	crush -d -c /path/to/project
+
+	# Run with custom data directory
+	crush -D /path/to/custom/.crush
+
+	# Print version
+	crush -v
 
-# Run with debug logging
-crush -d
+	# Run a single non-interactive prompt using --prompt flag
+	crush --prompt "Create a responsive React calculator component"
 
-# Run with debug logging in a specific directory
-crush -d -c /path/to/project
+	# Run a single non-interactive prompt using -p flag
+	crush -p "Explain the use of context in Go"
 
-# Run with custom data directory
-crush -D /path/to/custom/.crush
+	# Override model for this run (provider:model or model)
+	crush -p "Explain the use of context in Go" -m openai:gpt-4o
+	crush -p "Explain the use of context in Go" -m gpt-4o
 
-# Print version
-crush -v
+	# Run with prompt and quiet mode
+	crush -p "Generate a README for this project" -q
 
-# Run a single non-interactive prompt
-crush run "Explain the use of context in Go"
+	# Run in dangerous mode with prompt
+	crush -p "Create a simple API server" -y
 
-# Run in dangerous mode (auto-accept all permissions)
-crush -y
+	# Run a single non-interactive prompt (alternative method)
+	crush run "Explain the use of context in Go"
+
+	# Run in dangerous mode (auto-accept all permissions)
+	crush -y
   `,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		// If a prompt is provided via flag, run non-interactive and exit
+		prompt, _ := cmd.Flags().GetString("prompt")
+		if prompt != "" {
+			return handlePromptFlag(cmd, prompt)
```

Comment:
i think this feature could be useful for none interactive mode also (even more so than the tui) so running this after the model selection would be better.